src/main/java/fi/aalto/cs/apluscourses/model/APlusAuthentication.java

```diff
@@ -38,8 +73,18 @@ public void clear() {
    * @return True, if strings are equal, otherwise false.
    */
   public synchronized boolean tokenEquals(@Nullable String string) {
```

Comment:
Seems like this method had double-`synchronized` already in my implementation... 😬 My bad.

Fortunately, neither of them is no longer required.

Reply:
relying on the safety of IJ's implementation where it comes to credential retrieval :rocket: